assets/js/components/event/ControlPanel.tsx

```diff
@@ -109,127 +99,83 @@ const SettingsModal = ({ isOpen, onClose, elements }: SettingsModalProps) => {
   );
 };
 
-const stopBeingPresenter = (
-  eventChannel: Channel | undefined,
-  client: Client,
-  setClientStatus: React.Dispatch<React.SetStateAction<ClientStatus>>
+const getDropdownButton = (
+  key: string,
+  device: UseMicrophoneResult<TrackMetadata> | UseCameraResult<TrackMetadata>
 ) => {
-  setClientStatus("disconnected");
-  eventChannel?.push("presenter_remove", { email: client.email });
-  sessionStorageUnsetIsPresenter();
+  return (
+    <DropdownButton
+      key={key}
+      mainText={`${device.deviceInfo?.kind} source`}
+      currentSourceName={device.deviceInfo?.label}
+      sources={device.devices || []}
+      onSelectSource={(deviceId) => {
+        device.start(deviceId);
+      }}
+    />
+  );
+};
```

Comment:
It's a component, so you could use JSX to emphasise it:

```ts
const DeviceButton = ({ key, device }: {
  key: string,
  device: UseMicrophoneResult<TrackMetadata> | UseCameraResult<TrackMetadata>
}) => <DropdownButton
  key={key}
  mainText={`${device.deviceInfo?.kind} source`}
  currentSourceName={device.deviceInfo?.label}
  sources={device.devices || []}
  onSelectSource={(deviceId) => {
    device.start(deviceId);
  }}
/>
```

and then use it like this:

```ts
<SettingsModal
              isOpen={isOpen}
              onClose={onClose}
              elements={[
                <DeviceButton key={"1"} device={microphone}/>,
                <DeviceButton key={"2"} device={camera}/>
              ]}
            />

```